src/Kmail.ts

```diff
@@ -0,0 +1,223 @@
+import "core-js/features/object/entries";
+
+import {
+  extractItems,
+  extractMeat,
+  isGiftable,
+  toInt,
+  visitUrl,
+} from "kolmafia";
+import { arrayToCountedMap, chunk } from "./utils";
+
+type RawKmail = {
+  id: string;
+  type: string;
+  fromid: string;
+  azunixtime: string;
+  message: string;
+  fromname: string;
+  localtime: string;
+};
+
+export default class Kmail {
+  id: number;
+  date: Date;
+  type: "normal";
+  senderId: number;
+  senderName: string;
+  message: string;
+
+  /**
+   * Parses a kmail from KoL's native format
+   * 
+   * @param rawKmail Kmail in the format supplies by api.php
+   * @returns Parsed kmail
+   */
+  static parse(rawKmail: RawKmail): Kmail {
+    return new Kmail(rawKmail);
+  }
+
+  /**
+   * Returns all of the player's kmails
+   * 
+   * @returns Parsed kmails
+   */
+  static inbox(): Kmail[] {
+    return (JSON.parse(
+      visitUrl("api.php?what=kmail&for=ASSistant")
+    ) as RawKmail[]).map(Kmail.parse);
+  }
+
+  /**
+   * Bulk delete kmails
+   * 
+   * @param kmails Kmails to delete
+   * @returns Number of kmails deleted
+   */
+  static delete(kmails: Kmail[]): number {
+    const results = visitUrl(`messages.php?the_action=delete&box=Inbox&pwd&${kmails
+      .map((k) => `sel${k.id}=on`)
+      .join("&")}`);
+
+    return Number(results.match(/<td>(\d) messages? deleted.<\/td>/)?.[1] ?? 0);
+  }
+
+  private static _genericSend(
+    to: string | number,
+    message: string,
+    items: Map<Item, number> | Item[],
+    meat: number,
+    chunkSize: number,
```

Comment:
is it worth setting chunkSize's default value to 11 and/or verifying that you don't put a chunk size greater than 11 in there?